procurement_service/models/product_product.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2017 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import models, api
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    def _is_service_buy_make_to_order(self):
+        for product in self:
+            if (product.type == 'service' and len(product.route_ids) == 2 and
+                self.env.ref('stock.route_warehouse0_mto').id in
+                product.route_ids.ids and
+                self.env.ref('purchase.route_warehouse0_buy').id in
```

Comment:
I'm not sure about this, is there a reason to require both routes in the service definition? I tend to think that just checking that the service has at least one route will be enough here and it will allow other implementations to reuse this module.

Reply:
This is compliant with the module as it was in 8.0, so it doesn't break inheriting modules.

Reply:
again, forget about v8, which are your thought about this condition? I'm not saying it is not a good one, I'm just not sure about it.

Reply:
In this module we are trying to make services mimic the behaviour of stockable products, and as far as I know (and please correct me if I'm wrong, I'm still quite new to OdooWorld):
1. when a stockable product is sold and it has 'buy' route, a purchase order is made unless you have enough in inventory
2. when a stockable product is sold and it has 'mto' route, a purchase order is created

Since for services has no sense talking about inventory, both 1. and 2. are the same.
So yes I can change this and make it return true if any of the two routes is checked.

Reply:
Yes, that's how it works but the thing is this is the expression that defines which services will need a procurement. I was trying to see this module as start or base for others that use services in procurements. In this sense you could find interesting to have a look at: https://github.com/OCA/purchase-workflow/tree/10.0/subcontracted_service and https://github.com/Eficent/ao-odoo/tree/9.0/mrp_production_service. 

For the last one I was considerating to propose it to OCA and depend on this module. But maybe, I'm just over complicating it...

Reply:
I think those modules might depend on this, but they should override method `_need_procurement` and not `_is_service_buy_make_to_order`, because the latter defines the way _this_ module decides when services have to be managed differently. 
Actually I'm realizing that since this method is called also in other models it is not very inheriting-friendly, I'll make a new commit later that might help it.

Reply:
@lreficent I've committed, now the new method name is more 'anonymous' so it can be easily overridden

Reply:
Great, Thanks @SimoRubi